From 28e23cc7335935ba325f9ec5af492610ade20945 Mon Sep 17 00:00:00 2001
From: Tommy Chen <tommyyl.chen@mediatek.com>
Date: Fri, 16 Jun 2023 17:59:35 +0800
Subject: [PATCH] GENIO: uboot: bootlogo: remove video device before enter kernel

since uboot could not sync with kernel for now, remove video device
before entering kernel to avoid situations could not handled by kernel

Change-Id: I0d30a513892d1b5d1e9d6398c40ccd9da7cf6d76
Signed-off-by: Tommy Chen <tommyyl.chen@mediatek.com>
---

diff --git a/boot/bootm.c b/boot/bootm.c
index e3233fd..1f93a86 100644
--- a/boot/bootm.c
+++ b/boot/bootm.c
@@ -794,6 +794,14 @@
 		return ret;
 	}
 
+	if (IS_ENABLED(CONFIG_DM_VIDEO)) {
+		struct udevice *dev;
+
+		ret = uclass_first_device_err(UCLASS_VIDEO, &dev);
+		if (!ret)
+			device_remove(dev, 1);
+	}
+
 	/* Now run the OS! We hope this doesn't return */
 	if (!ret && (states & BOOTM_STATE_OS_GO))
 		ret = boot_selected_os(argc, argv, BOOTM_STATE_OS_GO,
